pkg/virt-controller/watch/vm/vm_test.go

```diff
@@ -7552,7 +7552,8 @@ var _ = Describe("VirtualMachine", func() {
 				Reason: reason,
 			}
 		}
-		DescribeTable("", func(vm *v1.VirtualMachine, vmi *v1.VirtualMachineInstance, expectedVolumeMigrationState *v1.VolumeMigrationState, expectCond bool) {
+		DescribeTable("and UpdateVolumesStrategy set to Migration", func(vm *v1.VirtualMachine, vmi *v1.VirtualMachineInstance, expectedVolumeMigrationState *v1.VolumeMigrationState, expectCond bool) {
+			vm.Spec.UpdateVolumesStrategy = pointer.P(v1.UpdateVolumesStrategyMigration)
```

Comment:
Are the new cases covered in the unit tests (condition removed when UpdateVolumesStrategyMigration is unset, condition added when VMI is final)?

With sufficient unit test coverage maybe we can get rid of the functional test?

Reply:
The functional test is more to demonstrate that we can recover from this situation. Initially, this PR should only have been a functional tests, but then it spotted this bugs as well :D.
So, no, the functional test in on purpose there.